### PR TITLE
sandbox/cgroup: add tracking helper to confirm transient scopes

### PR DESCRIPTION
**Pull request by ZeyadYasser**
`ConfirmSystemdAppTracking` can be used if the process is a member of an existing transient scope to avoid leaking cgroups in case of retries.

This PR is a building block for https://github.com/snapcore/snapd/pull/13411 in an attempt to split it into smaller chunks.
This will be later used in "snap run" to make it aware of inhibited snaps.

### CODE REVIEW

**Comment by ZeyadYasser** [comment]
Tests are failing due to a bug in `WaitWhileInhibited`, a fix is out for this https://github.com/snapcore/snapd/pull/13469.

**Review by olivercalder — APPROVED** [review]
LGTM, thank you!

File: sandbox/cgroup/tracking.go [hunk 1]
```diff
@@ -151,6 +152,24 @@ tryAgain:
...
+//
+// If the application process is not tracked then ErrCannotTrackProcess is returned.
+func ConfirmSystemdAppTracking(securityTag string) error {
+	pid := osGetpid()
+	path, err := cgroupProcessPathInTrackingCgroup(pid)
+	if err != nil {
+		return err
+	}
+
+	regex := regexp.MustCompile(fmt.Sprintf("%s-.+\\.scope$", securityTag))
```
Comment (bboozzoo) [inline sandbox/cgroup/tracking.go:165]: MustCompile would panic in the event that securityTag produced invalid format. Perhaps a better check would be to do:

```
// the transient scope of the application carries the security tag, eg:
// snap.hello-world.sh-4706fe54-7802-4808-aa7e-ae8b567239e0.scope
if strings.HasPrefix(filepath.Base(path), securityTag + "-")) {
    ...
}
```

Comment (ZeyadYasser) [reply sandbox/cgroup/tracking.go]: great catch, thank you!

updated!

**Review by bboozzoo — APPROVED** [review]
LGTM

**Comment by ZeyadYasser** [comment]
Since this PR only adds a helper function that is not used yet, I think it would be safe to skip spread tests, no?

**Review by pedronis — APPROVED** [review]
thanks

**Pull request merged**